src/main/java/hudson/remoting/JarLoaderImpl.java

```diff
@@ -1,28 +1,19 @@
 package hudson.remoting;
 
-import hudson.remoting.forward.Forwarder;
-
 import java.io.File;
 import java.io.IOException;
 import java.io.OutputStream;
 import java.io.Serializable;
 import java.net.URL;
-import java.security.DigestOutputStream;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
-import java.util.concurrent.ConcurrentSkipListSet;
 
 /**
  * Implements {@link JarLoader} to be called from the other side.
  *
- * TODO: move {@link #knownJars} and {@link #checksums} to another class to share it across
- * {@link JarLoaderImpl}s.
- *
```

Comment:
Sorry - this was meant only for https://github.com/jenkinsci/remoting/pull/43 - reverting.